__tests__/index.js

```diff
@@ -245,16 +245,55 @@ describe('Compress', () => {
       .expect(200, done)
   })
 
-  it('should not crash if no accept-encoding is sent', (done) => {
+  it('should not compress if no accept-encoding is sent (with the default)', (done) => {
     const app = new Koa()
+    app.use(compress({
+      threshold: 0
+    }))
+    app.use((ctx) => {
+      ctx.type = 'text'
+      ctx.body = buffer
+    })
+    server = app.listen()
 
-    app.use(compress())
-    app.use(sendBuffer)
+    request(server)
+      .get('/')
+      .set('Accept-Encoding', '')
```

Comment:
Ah interesting workaround.

Reply:
`unset()` didn't work, but setting the empty value did the trick.